File: csharp/LeetCode/ArraysAndHashing/PalindromeNumber.cs
namespace LeetCode.ArraysAndHashing
{
    public static class PalindromeNumber
    {
        public static bool IsPalindrome(int x)
        {
            if (x < 0)
            {
                return false;
            }

            if (x % 10 == 0 && x != 0)
            {
                return false;
            }

            string numStr = x.ToString();

            int left = 0;
            int right = numStr.Length - 1;

            while (left < right)
            {
                if (numStr[left] != numStr[right])
                {
                    return false;
                }

                left++;
                right--;
            }
            
            return true;
        }
    }
}

File: scala/src/main/scala/leetcode/ArraysAndHashing/PalindromeNumber.scala
package leetcode.ArraysAndHashing

object PalindromeNumber {
    def isPalindrome(x: Int): Boolean = {
        if (x < 0) {
            false
        } else if (x % 10 == 0 && x != 0) {
            false
        } else {
            val numStr = x.toString

            def checkPalindrome(left: Int, right: Int): Boolean = {
                if (left >= right) {
                    true
                } else if (numStr(left) != numStr(right)) {
                    false
                } else {
                    checkPalindrome(left + 1, right - 1)
                }
            }
            checkPalindrome(0, numStr.length - 1)
        }
    }
}